core/src/test/java/org/ehcache/config/ResourcePoolsImplTest.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.config;
+
+import static java.util.Arrays.asList;
+import java.util.Collection;
+import static org.ehcache.config.ResourcePoolsImpl.validateResourcePools;
+import static org.ehcache.config.ResourceType.Core.HEAP;
+import static org.ehcache.config.ResourceType.Core.OFFHEAP;
+import static org.ehcache.config.units.EntryUnit.ENTRIES;
+import static org.ehcache.config.units.MemoryUnit.KB;
+import static org.ehcache.config.units.MemoryUnit.MB;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import org.junit.Test;
+
+/**
+ *
+ * @author cdennis
+ */
+public class ResourcePoolsImplTest {
```

Comment:
I would have expected a few tests that make sure mixing entries and memory doesn't throw an exception and that everything succeeds when there's no pool inversion